packages/account/src/Sections/Profile/PersonalDetails/__tests__/personal-details.spec.tsx

```diff
@@ -0,0 +1,218 @@
+    WS: {
+        wait: (...payload: []) => Promise.resolve([...payload]),
+    },
+    useWS: () => undefined,
+}));
+
+describe('<PersonalDetailsForm />', () => {
+    const history = createBrowserHistory();
+
+    const promise = Promise.resolve();
+    const fetchResidenceList = jest.fn(() => promise);
+    const fetchStatesList = jest.fn(() => promise);
+    const residence_list = [
+        {
+            text: 'Text',
+            value: 'value',
+        },
+    ];
+    const mock_store = mockStore({
+        client: {
+            account_settings: {
+                first_name: 'John',
+                place_of_birth: 'Thailand',
+                citizen: 'Thailand',
+                email_consent: 1,
+            },
+            states_list: residence_list,
+            residence_list,
+            has_residence: true,
+            fetchResidenceList,
+            fetchStatesList,
+        },
+    });
+
+    const renderComponent = (modified_store = mock_store) => {
+        return render(
+            <Router history={history}>
+                <StoreProvider store={modified_store}>
+                    <PersonalDetailsForm history={history} />
+                </StoreProvider>
+            </Router>
+        );
+    };
+
+    it('should render successfully', async () => {
+        renderComponent();
+        expect(screen.queryByText(/Loading/)).not.toBeInTheDocument();
+        await waitFor(() => {
+            expect(
+                screen.getByText(
+                    /Please make sure your information is correct or it may affect your trading experience./i
+                )
+            ).toBeInTheDocument();
+        });
+    });
+
+    it('should render all the personal details fields', () => {
+        renderComponent();
+        const fields = [
+            'First name*',
+            'Last name*',
+            'Date of birth*',
+            'Country of residence*',
+            'Phone number*',
+            'First line of address*',
+            'Second line of address (optional)',
+            'Town/City*',
+            'State/Province (optional)',
+            'Postal/ZIP code',
+        ];
+        fields.forEach(value => {
+            expect(screen.getByText(value)).toBeInTheDocument();
+        });
+    });
+
+    it('should have "required" validation errors on required form fields', async () => {
+        renderComponent();
+
+        await waitFor(() => {
+            const first_name = screen.getByTestId('dt_first_name');
+            userEvent.clear(first_name);
+            expect(screen.getByText(/First name is required./)).toBeInTheDocument();
+        });
+    });
+
+    it('should display error for 2-50 characters length validation, for First name when entered characters are less than 2', async () => {
+        renderComponent();
+        await waitFor(() => {
+            const last_name = screen.getByTestId('dt_last_name');
+            fireEvent.input(last_name, { target: { value: 'b' } });
```

Comment:
We already have userevent as a dependency in this file. Cant we user userEvent.type(last_name, 'b') here and get rid of fireevent?